<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog_channel</class>
 <widget class="QDialog" name="Dialog_channel">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>536</width>
    <height>535</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>536</width>
    <height>535</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>536</width>
    <height>535</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Channel</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>-194</y>
        <width>502</width>
        <height>660</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label">
           <property name="toolTip">
            <string>Identifier used through the system to identify a given channel.</string>
           </property>
           <property name="text">
            <string>channel_id</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_channel_id">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="toolTip">
            <string>Numeric value to determine the order in which a channel will be processed. Channels will be processed in ascending order.</string>
           </property>
           <property name="text">
            <string>processing_order</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_processing_order">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>60000</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="toolTip">
            <string>Specifies the maximum number of data events to process within a batch for this channel.
</string>
           </property>
           <property name="text">
            <string>max_batch_size</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_max_batch_size">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>60000</number>
           </property>
           <property name="value">
            <number>1000</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="toolTip">
            <string>Specifies the maximum number of batches to send for a given channel during a 'synchronization' between two nodes. A 'synchronization' is equivalent to a push or a pull. For example, if there are 12 batches ready to be sent for a channel and max_batch_to_send is equal to 10, then only the first 10 batches will be sent even though 12 batches are ready.</string>
           </property>
           <property name="text">
            <string>max_batch_to_send</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_max_batch_to_send">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>60000</number>
           </property>
           <property name="value">
            <number>60</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="toolTip">
            <string>Specifies the maximum number of data rows to route for a channel at a time.</string>
           </property>
           <property name="text">
            <string>max_data_to_route</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_max_data_to_route">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>100000</number>
           </property>
           <property name="value">
            <number>100000</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>extract_period_millis</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_extract_period_millis">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>100000</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QLabel" name="label_7">
           <property name="toolTip">
            <string>Indicates whether the channel is enabled or disabled. If a channel is disabled, data is still captured for changes that occur on the source system, but it will not be routed and sent to the target until the channel is re-enabled.</string>
           </property>
           <property name="text">
            <string>enabled</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_enabled">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>1</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QLabel" name="label_9">
           <property name="toolTip">
            <string>Indicates if the old data will be included for routing. Routing can then use this data for processing. Defaults to true.</string>
           </property>
           <property name="text">
            <string>use_old_data_to_route</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_use_old_data_to_route">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>1</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QLabel" name="label_10">
           <property name="toolTip">
            <string>Indicates if the current data will be included for routing. Routing can then use this data for processing. Defaults to true.
</string>
           </property>
           <property name="text">
            <string>use_row_data_to_route</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_use_row_data_to_route">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>1</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_10">
         <item>
          <widget class="QLabel" name="label_11">
           <property name="toolTip">
            <string>Indicates if the primary key data will be include for routing. For example maybe a store ID is needed to apply logic on before sending to the appropriate target nodes. Defaults to true.</string>
           </property>
           <property name="text">
            <string>use_pk_data_to_route</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_use_pk_data_to_route">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>1</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_11">
         <item>
          <widget class="QLabel" name="label_12">
           <property name="toolTip">
            <string>Indicates whether a channel is available for initial loads and reverse initial loads.</string>
           </property>
           <property name="text">
            <string>reload_flag</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_reload_flag">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>1</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_12">
         <item>
          <widget class="QLabel" name="label_13">
           <property name="toolTip">
            <string>Indicates whether a channel is available for file synchronization.</string>
           </property>
           <property name="text">
            <string>file_sync_flag</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_file_sync_flag">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>1</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_13">
         <item>
          <widget class="QLabel" name="label_14">
           <property name="toolTip">
            <string>Indicates whether the channel contains big lobs. Some databases have shortcuts that SymmetricDS can take advantage of if it knows that the lob columns in SYM_DATA aren’t going to contain large lobs. The definition of how large a 'big' lob is varies from database to database.</string>
           </property>
           <property name="text">
            <string>contains_big_lob</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_contains_big_lob">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>1</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_14">
         <item>
          <widget class="QLabel" name="label_15">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Batching is the grouping of data, by channel, to be transferred and committed at the client together.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;table border=&quot;0&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px;&quot; cellspacing=&quot;2&quot; cellpadding=&quot;0&quot;&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Default&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;All changes that happen in a transaction are guaranteed to be batched together. Multiple transactions will be batched and committed together until there is no more data to be sent or the max_batch_size is reached.&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Transactional&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;Batches will map directly to database transactions. If there are many small database transactions, then there will be many batches. The max_batch_size column has no effect.&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Nontransactional&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;Multiple transactions will be batched and committed together until there is no more data to be sent or the max_batch_size is reached. The batch will be cut off at the max_batch_size regardless of whether it is in the middle of a transaction.&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;/table&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>batch_algorithm</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_batch_algorithm">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <item>
            <property name="text">
             <string>default</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>transactional</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>nontransactional</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_15">
         <item>
          <widget class="QLabel" name="label_16">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Determines how data will be loaded into the target tables. These are used during an initial load or a reverse initial load. Data loaders do not always have to load into the target relational database. They can write to a file, a web service, or any other type of non-relational data source. Data loaders can also use other techniques to increase performance of data loads into the target relation database.&lt;br/&gt;&lt;/p&gt;&lt;table border=&quot;0&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px;&quot; cellspacing=&quot;2&quot; cellpadding=&quot;0&quot;&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;default&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;Performs an insert first and if this fails will fall back to an update to load the data.&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;ftp_localhost&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;Sends the data in CSV format to a configured ftp location. These locations are setup in the TODO {SYM_HOME}/conf/ftp-extensions.xml&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;bulk&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;Assigns the appropriate bulk loader to this channel. Supported bulk loaders include: Microsoft SQL, PostgreSQL, MySQL and Amazon Redshift over S3.&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;mongodb&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;MongoDB data loader.&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;/table&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>data_loader_type</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_data_loader_type">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <item>
            <property name="text">
             <string>default</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>ftp_localhost</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>bulk</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>mongodb</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_16">
         <item>
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>description</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_description">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_17">
         <item>
          <widget class="QLabel" name="label_18">
           <property name="toolTip">
            <string>Determines a queue that the channel will sync in. Channels with the same queue name are processed synchronously (one at a time) and channels on different queues are processed asynchronously (in parallel).</string>
           </property>
           <property name="text">
            <string>queue</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_queue">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="text">
            <string>default</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_18">
         <item>
          <widget class="QLabel" name="label_19">
           <property name="toolTip">
            <string>Specifies the maximum network transfer rate in kilobytes per second. Use zero to indicate unlimited. When throttling the channel, make sure the channel is on its own queue or within a queue of channels that are throttled at the same rate. This is currently only implemented when staging is enabled.</string>
           </property>
           <property name="text">
            <string>max_network_kbps</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_max_network_kbps">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>28</height>
            </size>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_19">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="minimumSize">
        <size>
         <width>211</width>
         <height>41</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>211</width>
         <height>41</height>
        </size>
       </property>
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
